pydemolib/nessiedemo/flink.py

```diff
@@ -0,0 +1,173 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 Dremio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""NessieDemoFlink handles setting up Flink and Iceberg related objects."""
+
+import os
+import re
+import shutil
+import site
+from types import TracebackType
+from typing import Any, Tuple, TypeVar
+
+from pyflink.datastream import StreamExecutionEnvironment
+from pyflink.table import StreamTableEnvironment
+
+
+from .demo import _Util, NessieDemo
+
+T = TypeVar("T", bound="NessieDemoFlink")
+
+
+class NessieDemoFlink:
+    """`NessieDemoFlink` is a helper class for Flink in Nessie-Demos."""
+
+    __demo: NessieDemo
+
+    __env: StreamExecutionEnvironment
+    __jvm: Any
+
+    def __init__(self: T, demo: NessieDemo) -> None:
+        """Creates a `NessieDemoFlink` instance for respectively using the given `NessieDemo` instance."""
+        self.__demo = demo
+
+        if "hadoop" in self.__demo._get_versions_dict() and "tarball" in self.__demo._get_versions_dict()["hadoop"]:
```

Comment:
Nit: can you refactor this one to a `__build_hadoop_dir` (or a better name) function?